firebase-firestore/ktx/src/main/kotlin/com/google/firebase/firestore/ktx/Firestore.kt

```diff
@@ -163,3 +168,51 @@ class FirebaseFirestoreKtxRegistrar : ComponentRegistrar {
     override fun getComponents(): List<Component<*>> =
             listOf(LibraryVersionComponent.create(LIBRARY_NAME, BuildConfig.VERSION_NAME))
 }
+
+/**
+ * Attach a snapshotListener to a DocumentReference and use it as a coroutine flow
+ * @param metadataChanges Indicates whether metadata-only changes
+ */
+fun DocumentReference.toFlow(metadataChanges: MetadataChanges = MetadataChanges.EXCLUDE) = callbackFlow {
+    val listener = addSnapshotListener(metadataChanges) { value, error ->
+        if (value != null && value.exists()) {
```

Comment:
Why the check for `exists()`? What if I deliberately add a snapshot listener to a document that does not yet exist or if I want to be notified about deletions? Shouldn't I receive all updates so that I can handle them in my application accordingly?

Reply:
Yes you're right. I think I did it following the main use case (receive updates about an existing document reference), but I agree this could be useful too 👍